src/core/client/admin/test/community/community-rtl.spec.tsx

```diff
@@ -0,0 +1,545 @@
+  await waitFor(() =>
+    expect(resolvers.Mutation!.updateUserModerationScopes!.called).toBe(true)
+  );
+});
+
+it("promote user role as a site moderator", async () => {
+  const siteModerator = users.moderators[1];
+  const user = users.commenters[0];
+  const resolvers = createResolversStub<GQLResolver>({
+    Query: {
+      viewer: () => siteModerator,
+      settings: () => settingsWithMultisite,
+    },
+    Mutation: {
+      promoteModerator: ({ variables }) => {
+        expectAndFail(variables).toMatchObject({
+          userID: user.id,
+          siteIDs: [sites[0].id],
+        });
+        const userRecord = pureMerge<typeof user>(user, {
+          moderationScopes: { scoped: true, sites: [sites[0]] },
+        });
+        return {
+          user: userRecord,
+        };
+      },
+    },
+  });
+  await createTestRenderer({
+    resolvers,
+  });
+
+  const userRow = await screen.findByTestId(`community-row-${user.id}`);
+  const changeRoleButton = within(userRow).getByLabelText("Change role");
+  userEvent.click(changeRoleButton);
+
+  const popup = within(userRow).getByLabelText(
+    "A dropdown to promote/demote a user to/from sites"
+  );
+
+  const siteModButton = within(popup).getByRole("button", {
+    name: "Site Moderator",
+  });
+  siteModButton.click();
+
+  await waitFor(() => screen.getByTestId("siteModeratorActions-modal"));
+  const assignButton = screen.getByRole("button", { name: "Assign" });
+  userEvent.click(assignButton);
+
+  await waitFor(() =>
+    expect(resolvers.Mutation!.promoteModerator!.called).toBe(true)
+  );
+});
+
+it("demote user role as a site moderator", async () => {
+  const siteModeratorViewer = users.moderators[1];
+  const siteModeratorUser = users.moderators[2];
+
+  const resolvers = createResolversStub<GQLResolver>({
+    Query: {
+      viewer: () => siteModeratorViewer,
+      settings: () => settingsWithMultisite,
+    },
+    Mutation: {
+      demoteModerator: ({ variables }) => {
+        expectAndFail(variables).toMatchObject({
+          userID: siteModeratorUser.id,
+          siteIDs: [sites[0].id],
+        });
+        const userRecord = pureMerge<typeof siteModeratorUser>(
+          siteModeratorUser,
+          {
+            moderationScopes: { scoped: false, sites: [] },
+          }
+        );
+        return {
+          user: userRecord,
+        };
+      },
+    },
+  });
+  await createTestRenderer({
+    resolvers,
+  });
+
+  const userRow = await screen.findByTestId(
+    `community-row-${siteModeratorUser.id}`
+  );
+  const changeRoleButton = within(userRow).getByLabelText("Change role");
+  userEvent.click(changeRoleButton);
+
+  const popup = within(userRow).getByLabelText(
+    "A dropdown to promote/demote a user to/from sites"
+  );
+  const siteModButton = within(popup).getByRole("button", {
+    name: "Remove my sites from moderator",
+  });
+  siteModButton.click();
+
+  await waitFor(() => screen.getByTestId("siteModeratorActions-modal"));
```

Comment:
I think we should update to `await screen.findByTestId("siteModeratorActions-modal");`

Reply:
👍